src/services/pcn-router/src/Router.cpp

```diff
@@ -775,6 +775,75 @@ void Router::addPorts(const std::string &name, const PortsJsonObject &conf) {
   add_port<PortsJsonObject>(name, conf);
   if (get_shadow())
     getPorts(name)->setMac(getPorts(name)->getMac());
+
+  // Creates function objects containing the callback functions
+  // to be called when there are notifications on one netdev
+  std::function<void(const std::string, const std::string)> f_ip = std::bind(&Router::iface_ip_notification,
+                                this, std::placeholders::_1, std::placeholders::_2);
+  std::function<void(const std::string, const std::string)> f_mac = std::bind(&Router::iface_mac_notification,
+                                this, std::placeholders::_1, std::placeholders::_2);
+
+  // Register the new port to IP and MAC notifications arriving from netdevs
+  get_port(name)->subscribe_event("IP_ADDRESS", f_ip);
+  get_port(name)->subscribe_event("MAC_ADDRESS", f_mac);
+}
+
+// Callback function for ip notification
+void Router::iface_ip_notification(const std::string name_iface, const std::string info_ipAddress) {
+  std::lock_guard<std::mutex> guard(router_mutex);
+
+  // Check if the interface is connected to this cube
```

Comment:
Why do you need to check that the interface is connected to that cube?
If you receive a notification of an interface that is not connected to the cube it means there is a bug somewhere. IMO this check is useless.